Update ticket page

song50119 opened the pull request:
Resolves #387

song50119 committed:
Update ticket page

song50119 committed:
Display stacking record by tickets status

song50119 committed:
Add rewards earned layout

song50119 committed:
Calculate ticket statuses
- calculate total tickets live and add 5 tickets live recent
- calculate stacking record by ticket status
- add 5 tickets to RecentActivity

song50119 committed:
Handler purchase ticket
- Add modal to select wallet, account to purchase
- Add dummy VSP host to run test
- Add confirm purchase ticket
- Add VSP modal

song50119 committed:
VSP selector handler
- Save vsp list and remember host to config
- Validate vsp host before save
- Validate ticket quanity, vsp host, fee and total cost before purchase
- Fix bugs account selector page purchase

song50119 committed:
Get list VSP from API and filter by Network

song50119 committed:
Fix bug purchase and display live tickets recent

oshorefueled commented on go.sum:
```diff
@@ -8,8 +8,6 @@ decred.org/cspp v0.3.0/go.mod h1:UygjYilC94dER3BEU65Zzyoqy9ngJfWCD2rdJqvUs2A=
 decred.org/dcrwallet v1.6.0-rc4/go.mod h1:lsrNbuKxkPGeHXPufxNTckwQopCEDz0r3t0a8JCKAmU=
 dmitri.shuralyov.com/gpu/mtl v0.0.0-20190408044501-666a987793e9/go.mod h1:H6x//7gZCb22OMCxBHrMx7a5I7Hp++hsVxbQ4BYO7hU=
-gioui.org v0.0.0-20210228180843-e1248651c871 h1:LDHDc7aDPdzf9I5MfEnSRCSJj6W0Nf6Dl9cLOGj4/ew=
-gioui.org v0.0.0-20210228180843-e1248651c871/go.mod h1:Y+uS7hHMvku1Q+ooaoq6fYD5B2LGoT8JtFgvmYmRzTw=
```
There's no go.mod file committed on this PR, a go.sum shouldn't be committed either.

oshorefueled commented on ui/compontents.go:
```diff
@@ -473,6 +473,11 @@ func (page *pageCommon) accountSelectorLayout(gtx layout.Context, callingPage, s
 		return d(gtx, acctName, walName, bal, page.wallAcctSelector.receivingAccountBtn)
+	case callingPage == "purchase":
+		wallSelect := page.info.Wallets[page.wallAcctSelector.selectedPurchaseTicketWallet]
+		acctName := wallSelect.Accounts[page.wallAcctSelector.selectedPurchaseTicketAccount].Name
```
`wallSelect.Accounts[page.wallAcctSelector.selectedPurchaseTicketAccount]` can be assigned to a variable ass "acc". You can then access the name as `acc.Name` and balance as  `acc.TotalBalance`

oshorefueled commented on ui/page.go:
```diff
@@ -27,16 +27,43 @@ type pageIcons struct {
+	ticketExpiredIcon2,
+	ticketExpiredIcon3,
+	ticketRevokedIcon1,
+	ticketRevokedIcon2,
```
Why do we have icons repeated with integer increments in their names? They seem to be the same icons.

song50119 replied:
Same icon but different size

oshorefueled commented on wallet/commands.go:
```diff
@@ -934,41 +940,85 @@ func (wal *Wallet) SetupAccountMixer(walletID int, walletPassphrase string, errC
+				errChan <- err
+			}()
+			return
+		}
```
This looks like a new vspd instance is created every time the PurchaseTicket method is called.

song50119 committed:
Only create new VSPD when purchase form changed

song50119 committed:
Update ticket icons
- Remove icons integer increments
- Use one icon and scale on different layout section

song50119 committed:
Fix recent activity display wrong date
